sources/packages/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -214,13 +203,16 @@ export class ApplicationStudentsController extends BaseController {
       userToken.userId,
     );
 
-    await this.validateOverlappingDatesAndPIR(
+    await this.applicationService.validateOverlappingDatesAndPIR(
```

Comment:
Please move this method call should inside the try/catch to generate the proper HTTP exception.